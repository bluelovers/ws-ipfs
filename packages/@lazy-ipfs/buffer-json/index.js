"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringify = exports.parse = void 0;
function parse(dataBuffer) {
    return JSON.parse(Buffer.from(dataBuffer).toString());
}
exports.parse = parse;
function stringify(dataValue) {
    return Buffer.from(JSON.stringify(dataValue));
}
exports.stringify = stringify;
exports.default = {
    parse,
    stringify,
};
//# sourceMappingURL=index.js.map