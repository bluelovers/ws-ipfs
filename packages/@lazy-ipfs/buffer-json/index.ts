export function parse<T>(dataBuffer: Buffer): T
{
	return JSON.parse(Buffer.from(dataBuffer).toString())
}

export function stringify<T>(dataValue: T): Buffer
{
	return Buffer.from(JSON.stringify(dataValue))
}

export default {
	parse,
	stringify,
}
