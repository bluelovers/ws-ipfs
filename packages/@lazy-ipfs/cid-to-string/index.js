"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cidToString = void 0;
const tslib_1 = require("tslib");
const detect_cid_lib_1 = (0, tslib_1.__importDefault)(require("@lazy-ipfs/detect-cid-lib"));
const basics_1 = require("multiformats/basics");
function cidToString(cid, base) {
    const type = (0, detect_cid_lib_1.default)(cid);
    if (typeof base === 'undefined' || base === null) {
        return cid.toString();
    }
    else if (typeof base === 'string') {
        if (type === "@ipld/js-cid/CID" /* js_cids */) {
            return cid.toString(base);
        }
        base = basics_1.bases[base];
    }
    else if (type === "@ipld/js-cid/CID" /* js_cids */) {
        return cid.toString(base.name);
    }
    return cid.toString(base);
}
exports.cidToString = cidToString;
exports.default = cidToString;
//# sourceMappingURL=index.js.map