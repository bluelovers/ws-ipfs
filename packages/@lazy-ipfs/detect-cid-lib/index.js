"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.assertMultiformatsCID = exports.assertJsCID = exports.typeofCID = exports.isMultiformatsCID = exports.isJsCID = exports._isCIDLike = exports._isArrayLike = exports.SymbolMultiformatsCID = exports.SymbolJsCID = exports.EnumTypeofCID = void 0;
var EnumTypeofCID;
(function (EnumTypeofCID) {
    EnumTypeofCID["js_cid"] = "@ipld/js-cid/CID";
    EnumTypeofCID["multiformats_cid"] = "@ipld/js-multiformats/CID";
})(EnumTypeofCID = exports.EnumTypeofCID || (exports.EnumTypeofCID = {}));
exports.SymbolJsCID = Symbol.for("@ipld/js-cid/CID" /* js_cid */);
/**
 * @deprecated this is not exists
 * @see https://github.com/multiformats/js-multiformats/pull/109
 */
exports.SymbolMultiformatsCID = Symbol.for("@ipld/js-multiformats/CID" /* multiformats_cid */);
function _isArrayLike(input) {
    const type = typeof input;
    return Array.isArray(input) || type !== 'function' && type !== 'string' && typeof (input === null || input === void 0 ? void 0 : input.length) === 'number' && typeof input.forEach === 'function' && typeof input.slice === 'function';
}
exports._isArrayLike = _isArrayLike;
function _isCIDLike(cid) {
    return typeof (cid === null || cid === void 0 ? void 0 : cid.version) === 'number' && typeof cid.multihash !== 'undefined';
}
exports._isCIDLike = _isCIDLike;
function isJsCID(cid) {
    return (cid[exports.SymbolJsCID] === true) && _isArrayLike(cid.multihash);
}
exports.isJsCID = isJsCID;
function isMultiformatsCID(cid) {
    return !isJsCID(cid) && _isArrayLike(cid.bytes) && !_isArrayLike(cid.multihash);
}
exports.isMultiformatsCID = isMultiformatsCID;
function typeofCID(cid) {
    if (isJsCID(cid)) {
        return "@ipld/js-cid/CID" /* js_cid */;
    }
    else if (isMultiformatsCID(cid)) {
        return "@ipld/js-multiformats/CID" /* multiformats_cid */;
    }
}
exports.typeofCID = typeofCID;
function assertJsCID(cid) {
    if (!isJsCID(cid)) {
        throw new TypeError(`CID is not '${"@ipld/js-cid/CID" /* js_cid */}'`);
    }
}
exports.assertJsCID = assertJsCID;
function assertMultiformatsCID(cid) {
    if (!isMultiformatsCID(cid)) {
        throw new TypeError(`CID is not '${"@ipld/js-multiformats/CID" /* multiformats_cid */}'`);
    }
}
exports.assertMultiformatsCID = assertMultiformatsCID;
exports.default = typeofCID;
//# sourceMappingURL=index.js.map