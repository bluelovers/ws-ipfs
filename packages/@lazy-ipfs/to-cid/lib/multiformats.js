"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toMultiformatsCID = void 0;
const tslib_1 = require("tslib");
const err_code_1 = (0, tslib_1.__importDefault)(require("err-code"));
const multiformats_1 = require("multiformats");
const js_cids_1 = require("@lazy-ipfs/detect-cid-lib/lib/js-cids");
const js_multiformats_1 = require("@lazy-ipfs/detect-cid-lib/lib/js-multiformats");
const util_1 = require("@lazy-ipfs/detect-cid-lib/lib/util");
const parsePath_1 = require("@lazy-ipfs/parse-ipfs-path/lib/parsePath");
const _handleLibCID_1 = require("./_handleLibCID");
function toMultiformatsCID(cidInput, libCID) {
    libCID = (0, _handleLibCID_1._handleLibCID)(libCID, multiformats_1.CID);
    if (typeof cidInput === 'string') {
        return libCID.parse((0, parsePath_1.parsePath)(cidInput, {
            unsafeReturn: true,
            noThrow: true,
        }).hash);
    }
    else if ((0, util_1._isArrayLike)(cidInput)) {
        return libCID.decode(cidInput);
    }
    let cid = libCID.asCID(cidInput);
    if ((typeof cid === 'undefined' || cid === null)) {
        if ((0, js_cids_1.isRawJsCIDLike)(cidInput)) {
            cid = libCID.asCID((0, js_cids_1.toRawJsCIDFake)(cidInput));
        }
        else if ((0, js_multiformats_1.isRawMultiformatsCIDLike)(cidInput)) {
            cid = libCID.asCID((0, js_multiformats_1.toRawMultiformatsCIDFake)(cidInput));
        }
        else if ((0, parsePath_1.isParsePathResultLoose)(cidInput)) {
            cid = toMultiformatsCID(cidInput.hash);
        }
    }
    if (!cid) {
        throw (0, err_code_1.default)(new TypeError(`Invalid type for convert to MultiformatsCID`), {
            input: cidInput,
        });
    }
    return cid;
}
exports.toMultiformatsCID = toMultiformatsCID;
exports.default = toMultiformatsCID;
//# sourceMappingURL=multiformats.js.map