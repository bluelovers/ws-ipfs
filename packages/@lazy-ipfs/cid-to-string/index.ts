import { ICIDObject } from '@lazy-ipfs/detect-cid-lib/lib/types';
import typeofCID, { EnumTypeofCID } from '@lazy-ipfs/detect-cid-lib';
import JsCID from 'cids';
import { bases as basesMap } from 'multiformats/basics';
import { BaseCodec, Codec } from 'multiformats/types/bases/base';
import { CID as MultiformatsCID } from 'multiformats'

export type IBaseNameOrBaseCodec = Parameters<ICIDObject["toString"]>[0];

export type IBaseName = Extract<IBaseNameOrBaseCodec, string>;
export type IBaseCodec = Exclude<IBaseNameOrBaseCodec, string | undefined>;

export function cidToString(cid: ICIDObject, base?: IBaseNameOrBaseCodec)
{
	const type = typeofCID(cid);

	if (typeof base === 'undefined' || base === null)
	{
		return cid.toString()
	}
	else if (typeof base === 'string')
	{
		if (type === EnumTypeofCID.js_cids)
		{
			return (cid as JsCID).toString(base)
		}

		base = basesMap[base]
	}
	else if (type === EnumTypeofCID.js_cids)
	{
		return (cid as JsCID).toString((base as Codec<IBaseName, string>).name)
	}

	return (cid as MultiformatsCID).toString(base as IBaseCodec)
}

/**
 * default ipfs cid hash
 */
export function cidToQmHash(cid: ICIDObject)
{
	return cidToString(cid.toV0(), 'base58btc')
}

/**
 * use for subdomain
 */
export function cidToBase32(cid: ICIDObject)
{
	return cidToString(cid.toV1(), 'base32')
}

export default cidToString
