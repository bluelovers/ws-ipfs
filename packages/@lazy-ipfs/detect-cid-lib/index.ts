import { isMultiformatsCID, isRawMultiformatsCIDLike } from './lib/js-multiformats';
import { isJsCID, isRawJsCIDLike } from './lib/js-cids';
import err_code from 'err-code';

export { SymbolJsCID } from './lib/js-cids';

export * from './lib/js-cids';
export * from './lib/js-multiformats';
export * from './lib/util';
export * from './lib/types';

export const enum EnumTypeofCID
{
	/**
	 * for typo
	 * @deprecated
	 */
	js_cid = '@ipld/js-cid/CID',
	js_cids = '@ipld/js-cid/CID',
	multiformats_cid = '@ipld/js-multiformats/CID',
	js_multiformats = '@ipld/js-multiformats/CID',
	js_multiformat = '@ipld/js-multiformats/CID',
}

export function typeofCID(cid: unknown, throwError?: boolean)
{
	if (isMultiformatsCID(cid))
	{
		return EnumTypeofCID.multiformats_cid
	}
	else if (isJsCID(cid))
	{
		return EnumTypeofCID.js_cids
	}
	else if (throwError)
	{
		throw err_code(new TypeError(`Unknown type of cid`), {
			input: cid,
		})
	}
}

export function typeofRawCID(cid: unknown, throwError?: boolean)
{
	if (isRawMultiformatsCIDLike(cid))
	{
		return EnumTypeofCID.multiformats_cid
	}
	else if (isRawJsCIDLike(cid))
	{
		return EnumTypeofCID.js_cids
	}
	else if (throwError)
	{
		throw err_code(new TypeError(`Unknown type of raw cid`), {
			input: cid,
		})
	}
}

export default typeofCID
