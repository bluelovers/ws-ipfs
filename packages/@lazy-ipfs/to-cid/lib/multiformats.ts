import err_code from 'err-code';
import { CID as MultiformatsCID } from 'multiformats'
import { isRawJsCIDLike, SymbolJsCID, toRawJsCIDFake } from '@lazy-ipfs/detect-cid-lib/lib/js-cids';
import { ICIDValueInput, ICIDValueOrRaw } from '@lazy-ipfs/detect-cid-lib/lib/types';
import { IRawMultiformatsCIDFake, isRawMultiformatsCIDLike, toRawMultiformatsCIDFake } from '@lazy-ipfs/detect-cid-lib/lib/js-multiformats';
import { _isArrayLike } from '@lazy-ipfs/detect-cid-lib/lib/util';
import { parsePath } from '@lazy-ipfs/parse-ipfs-path/lib/parsePath';
import { _handleLibCID } from './_handleLibCID';
import { EnumTypeofCID } from '@lazy-ipfs/detect-cid-lib';

export function toMultiformatsCID<T extends ICIDValueInput, C extends MultiformatsCID = MultiformatsCID>(cidInput: T, libCID?: Pick<typeof MultiformatsCID, 'parse' | 'decode' | 'asCID'> | EnumTypeofCID): C
{
	libCID = _handleLibCID(libCID, MultiformatsCID);

	if (typeof cidInput === 'string')
	{
		return libCID.parse(parsePath(cidInput, {
			unsafeReturn: true,
			noThrow: true,
		}).hash) as any
	}
	else if (_isArrayLike(cidInput))
	{
		return libCID.decode(cidInput) as any
	}

	let cid = libCID.asCID(cidInput);

	if ((typeof cid === 'undefined' || cid === null))
	{
		if (isRawJsCIDLike(cidInput))
		{
			cid = libCID.asCID(toRawJsCIDFake(cidInput));
		}
		else if (isRawMultiformatsCIDLike(cidInput))
		{
			cid = libCID.asCID(toRawMultiformatsCIDFake(cidInput))
		}
	}

	if (!cid)
	{
		throw err_code(new TypeError(`Invalid type for convert to MultiformatsCID`), {
			input: cidInput,
		})
	}

	return cid as any
}

export default toMultiformatsCID
