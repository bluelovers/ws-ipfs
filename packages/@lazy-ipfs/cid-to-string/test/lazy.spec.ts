import JsCID from 'cids';
import { EnumTypeofCID } from '@lazy-ipfs/detect-cid-lib';
import toCID from '@lazy-ipfs/to-cid';
import { CID as MultiformatsCID } from 'multiformats'
import { cidToString } from '../index';
import { bases as basesMap } from 'multiformats/basics';

describe(`BaseName`, () =>
{
	const cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	test(EnumTypeofCID.js_cids, () =>
	{
		let co = toCID(cid, JsCID);

		let actual01 = cidToString(co.toV1(), 'base32');
		let actual02 = cidToString(co, 'base58btc');

		expect(actual01).toMatchSnapshot();
		expect(actual02).toMatchSnapshot();

	});

	test(EnumTypeofCID.multiformats_cid, () =>
	{
		let co = toCID(cid, MultiformatsCID);

		let actual01 = cidToString(co.toV1(), 'base32');
		let actual02 = cidToString(co, 'base58btc');

		expect(actual01).toMatchSnapshot();
		expect(actual02).toMatchSnapshot();

	});

})

describe(`BaseCodec`, () =>
{
	const cid = 'Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a';

	test(EnumTypeofCID.js_cids, () =>
	{
		let co = toCID(cid, JsCID);

		let actual01 = cidToString(co.toV1(), basesMap['base32']);
		let actual02 = cidToString(co, basesMap['base58btc']);

		expect(actual01).toMatchSnapshot();
		expect(actual02).toMatchSnapshot();

	});

	test(EnumTypeofCID.multiformats_cid, () =>
	{
		let co = toCID(cid, MultiformatsCID);

		let actual01 = cidToString(co.toV1(), basesMap['base32']);
		let actual02 = cidToString(co, basesMap['base58btc']);

		expect(actual01).toMatchSnapshot();
		expect(actual02).toMatchSnapshot();

	});

})
