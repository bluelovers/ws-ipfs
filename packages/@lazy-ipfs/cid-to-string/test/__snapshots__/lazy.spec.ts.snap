// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCodec @ipld/js-cid/CID 1`] = `"bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m"`;

exports[`BaseCodec @ipld/js-cid/CID 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`BaseCodec @ipld/js-multiformats/CID 1`] = `"bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m"`;

exports[`BaseCodec @ipld/js-multiformats/CID 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`BaseName @ipld/js-cid/CID 1`] = `"bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m"`;

exports[`BaseName @ipld/js-cid/CID 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;

exports[`BaseName @ipld/js-multiformats/CID 1`] = `"bafybeifx7yeb55armcsxwwitkymga5xf53dxiarykms3ygqic223w5sk3m"`;

exports[`BaseName @ipld/js-multiformats/CID 2`] = `"Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"`;
